lib/types/inet-address.js

```diff
@@ -10,238 +11,260 @@ const utils = require("../utils");
  * @param {Buffer} buffer
  * @constructor
  */
-function InetAddress(buffer) {
-    if (
-        !(buffer instanceof Buffer) ||
-        (buffer.length !== 4 && buffer.length !== 16)
-    ) {
-        throw new TypeError("The ip address must contain 4 or 16 bytes");
-    }
+class InetAddress {
+    #internal;
 
-    /**
-     * Immutable buffer that represents the IP address
-     * @type Array
-     */
-    this.buffer = buffer;
+    constructor(buffer) {
+        this.#internal = rust.InetAddressWrapper.new(buffer);
+    }
 
     /**
      * Returns the length of the underlying buffer
+     * @readonly
      * @type Number
      */
-    this.length = buffer.length;
+    get length() {
+        return this.#internal.getLength();
+    }
+
+    set length(value) {
+        throw new Error("InetAddress length is read-only");
+    }
 
     /**
      * Returns the Ip version (4 or 6)
+     * @readonly
      * @type Number
```

Comment:
Maybe `@returns {Number}`? 
Docs: https://jsdoc.app/tags-returns

Reply:
According to https://stackoverflow.com/a/22276207 @type {number} should be used